spec/models/source_spec.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Source, type: :model do
+  context 'when creating two Sources with different cases' do
+    it 'does NOT create a second source' do
+      described_class.find_or_create_by({ name: 'nimble' })
+
+      expect do
+        described_class.find_or_create_by({ name: 'nImBle' })
+      end.not_to change(described_class, :count)
+    end
+  end
+
+  context 'when creating two Sources with different names' do
+    it 'saves 2 sources' do
+      described_class.find_or_create_by({ name: 'nimble1' })
+
+      expect do
+        described_class.find_or_create_by({ name: 'nimble2' })
+      end.to change(described_class, :count).by(1)
+    end
+  end
```

Comment:
As already said, the description and the tests are different.

Either we adjust the test to suit with the description:

```suggestion
  context 'when creating two Sources with different names' do
    it 'saves 2 sources' do
      expect do
        described_class.find_or_create_by({ name: 'nimble1' })
        described_class.find_or_create_by({ name: 'nimble2' })
      end.to change(described_class, :count).by(2)
    end
  end
```

Or we adjust the description to suit with the test:

```suggestion
context 'given a source "nimble1"' do
  context 'when creating a second source "nimble2"' do
    it 'creates 1 new source' do
      described_class.find_or_create_by({ name: 'nimble1' })

      expect do
        described_class.find_or_create_by({ name: 'nimble2' })
      end.to change(described_class, :count).by(1)
    end
  end
end
```


The point is just: you cannot have a test `it saves 2` that `expect count change by 1` 🙈

Reply:
The first example makes a lot of senses. 👍